### ISSUE TITLE
Resizing Crashes on Mac OS X

### ISSUE THREAD

**Issue by Magnie** [kivy/kivent#99]
When attempting to resize the window (on Mac OS X Yosemite) of the examples provided (except for Example 1).

Example 1 (resize works):
```
$ kivy main.py 
[INFO   ] [Logger      ] Record log in /Applications/Kivy.app/Contents/Resources/.kivy/logs/kivy_15-11-03_19.txt
[INFO   ] [Kivy        ] v1.9.1-dev
[INFO   ] [Python      ] v2.7.10 (default, Jul 14 2015, 19:46:27) 
[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.39)]
[INFO   ] [Factory     ] 177 symbols loaded
[INFO   ] [Image       ] Providers: img_tex, img_imageio, img_dds, img_gif, img_pygame (img_pil, img_ffpyplayer ignored)
[INFO   ] [Window      ] Provider: pygame
[INFO   ] [GL          ] OpenGL version <2.1 INTEL-10.6.33>
[INFO   ] [GL          ] OpenGL vendor <Intel>
[INFO   ] [GL          ] OpenGL renderer <Intel(R) Iris(TM) Graphics 6100>
[INFO   ] [GL          ] OpenGL parsed version: 2, 1
[INFO   ] [GL          ] Shading version <1.20>
[INFO   ] [GL          ] Texture max size <16384>
[INFO   ] [GL          ] Texture max units <16>
[INFO   ] [Window      ] virtual keyboard not allowed, single mode, not docked
[INFO   ] [OSC         ] using <multiprocessing> for socket
[INFO   ] [KivEnt      ] Vertex Format: vertex_format_4f registered. Size per vertex is: 16. Format is [('pos', 2, 'float', 0, False), ('uvs', 2, 'float', 8, False)].
[INFO   ] [KivEnt      ] Vertex Format: vertex_format_7f registered. Size per vertex is: 28. Format is [('pos', 2, 'float', 0, False), ('uvs', 2, 'float', 8, False), ('rot', 1, 'float', 16, False), ('center', 2, 'float', 20, False)].
[INFO   ] [KivEnt      ] Vertex Format: vertex_format_4f4ub registered. Size per vertex is: 20. Format is [('pos', 2, 'float', 0, False), ('uvs', 2, 'float', 8, False), ('vColor', 4, 'ubyte', 16, True)].
[INFO   ] [KivEnt      ] Model Manager reserved space for vertex format: vertex_format_4f. 75 KiB was reserved for vertices, fitting a total of 4800. 25 KiB was reserved for indices fitting a total of 12800.
[INFO   ] [KivEnt      ] Model Manager reserved space for vertex format: vertex_format_7f. 75 KiB was reserved for vertices, fitting a total of 2742. 25 KiB was reserved for indices fitting a total of 12800.
[INFO   ] [KivEnt      ] Model Manager reserved space for vertex format: vertex_format_4f4ub. 75 KiB was reserved for vertices, fitting a total of 3840. 25 KiB was reserved for indices fitting a total of 12800.
[INFO   ] [KivEnt      ] We will need 684 KiB for game, we have 102400 KiB
[INFO   ] [Base        ] Start application main loop
[INFO   ] [Context     ] Reloading graphics data...
[INFO   ] [Context     ] Reloading done in 0.0392s
[INFO   ] [Context     ] Reloading graphics data...
[INFO   ] [Context     ] Reloading done in 0.0252s
[INFO   ] [Context     ] Reloading graphics data...
[INFO   ] [Context     ] Reloading done in 0.0229s
[INFO   ] [Base        ] Leaving application in progress...
```
Example 2 (resize doesn't work):
```
$ kivy main.py 
[INFO   ] [Logger      ] Record log in /Applications/Kivy.app/Contents/Resources/.kivy/logs/kivy_15-11-03_20.txt
[INFO   ] [Kivy        ] v1.9.1-dev
[INFO   ] [Python      ] v2.7.10 (default, Jul 14 2015, 19:46:27) 
[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.39)]
[INFO   ] [Factory     ] 177 symbols loaded
[INFO   ] [Image       ] Providers: img_tex, img_imageio, img_dds, img_gif, img_pygame (img_pil, img_ffpyplayer ignored)
[INFO   ] [Window      ] Provider: pygame
[INFO   ] [GL          ] OpenGL version <2.1 INTEL-10.6.33>
[INFO   ] [GL          ] OpenGL vendor <Intel>
[INFO   ] [GL          ] OpenGL renderer <Intel(R) Iris(TM) Graphics 6100>
[INFO   ] [GL          ] OpenGL parsed version: 2, 1
[INFO   ] [GL          ] Shading version <1.20>
[INFO   ] [GL          ] Texture max size <16384>
[INFO   ] [GL          ] Texture max units <16>
[INFO   ] [Window      ] virtual keyboard not allowed, single mode, not docked
[INFO   ] [OSC         ] using <multiprocessing> for socket
[INFO   ] [Text        ] Provider: pygame
[INFO   ] [Shader      ] Read </Users/josepha/Downloads/kivy-kivent-ca709d3/examples/assets/glsl/positionshader.glsl>
[ERROR  ] [GameSystem  ] System_id: renderer not attached retrying in 1 sec. If you see this error once or twice, we are probably just waiting on the KV file to load. If you see it a whole bunch something is probably wrong. Make sure all systems are setup properly.
[INFO   ] [Base        ] Start application main loop
[INFO   ] [GL          ] NPOT texture support is available
[INFO   ] [KivEnt      ] Vertex Format: vertex_format_4f registered. Size per vertex is: 16. Format is [('pos', 2, 'float', 0, False), ('uvs', 2, 'float', 8, False)].
[INFO   ] [KivEnt      ] Vertex Format: vertex_format_7f registered. Size per vertex is: 28. Format is [('pos', 2, 'float', 0, False), ('uvs', 2, 'float', 8, False), ('rot', 1, 'float', 16, False), ('center', 2, 'float', 20, False)].
[INFO   ] [KivEnt      ] Vertex Format: vertex_format_4f4ub registered. Size per vertex is: 20. Format is [('pos', 2, 'float', 0, False), ('uvs', 2, 'float', 8, False), ('vColor', 4, 'ubyte', 16, True)].
[INFO   ] [KivEnt      ] position allocated 128 KiB
[INFO   ] [KivEnt      ] Batches for canvas: <kivy.graphics.instructions.RenderContext object at 0x111dc51d0> will have 8192 verts and 65536 indices.VBO will be 128 in KiB per frame with 20 total vbos, an estimated 2048 enities fit in each batch with 4 verts per entity
[INFO   ] [KivEnt      ] renderer allocated 6292 KiB
[INFO   ] [KivEnt      ] Model Manager reserved space for vertex format: vertex_format_4f. 75 KiB was reserved for vertices, fitting a total of 4800. 25 KiB was reserved for indices fitting a total of 12800.
[INFO   ] [KivEnt      ] Model Manager reserved space for vertex format: vertex_format_7f. 75 KiB was reserved for vertices, fitting a total of 2742. 25 KiB was reserved for indices fitting a total of 12800.
[INFO   ] [KivEnt      ] Model Manager reserved space for vertex format: vertex_format_4f4ub. 75 KiB was reserved for vertices, fitting a total of 3840. 25 KiB was reserved for indices fitting a total of 12800.
[INFO   ] [KivEnt      ] We will need 6848 KiB for game, we have 102400 KiB
[INFO   ] [Context     ] Reloading graphics data...
[INFO   ] [Context     ] Reloading done in 0.0266s
[ERROR  ] OpenGL Error (Renderer._set_blend_func-glBlendFunc) 1282 / 502
[ERROR  ] OpenGL Error (FixedVBO.update_buffer-glBufferSubData) 1281 / 501
[ERROR  ] OpenGL Error (FixedVBO.update_buffer-glBufferSubData) 1281 / 501
Fatal Python error: (pygame parachute) Segmentation Fault
Abort trap: 6
```
Is there any other information you need?

**Comment by MojoMateo** [comment]
same problem for me

**Comment by Kovak** [comment]
This crash I believe is fixed in the master version of Kivy, can you check?

Also you should consider upgrading to the SDL2 providers, pygame has been deprecated at this point in time.

**Comment by MojoMateo** [comment]
`[INFO   ] [Kivy        ] v1.9.1
[INFO   ] [Python      ] v3.5.1 (default, Apr 18 2016, 11:46:32)
[GCC 4.2.1 Compatible Apple LLVM 7.3.0 (clang-703.0.29)]`

I can't find any related issues on the kivy project.

I'll try SDL2 privders

**Comment by Kovak** [comment]
In order to try a master kivy you must clone from github and install from source. Either that or wait for the next release.

**Comment by MojoMateo** [comment]
You were right the problem is caused by pygame.

I used `kivy setup.py build_ext install` instead of `python setup.py build_ext install`
to be sure the installation goes to the Kivy.app which is the recommanded osx install and this one is bundeled with the SDL2 lib instead of pygame.

**Comment by Kovak** [comment]
Just a note here also: There is another issue also
The

[ERROR  ] OpenGL Error (FixedVBO.update_buffer-glBufferSubData) 1281 / 501

stuff is fixed by:

https://github.com/kivy/kivy/pull/3970

but this has not been merged yet
